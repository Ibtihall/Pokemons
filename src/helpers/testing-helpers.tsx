import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import store from '../store';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { renderHook } from '@testing-library/react-hooks';

export const Wrapper: React.FC = (props:any): JSX.Element => {
  return (
    <Provider store={store}>
      {//@ts-ignore
      <Router>
        {props.children} 
      </Router>}
    </Provider>
  );
};


export const renderWithIntl = (
  ui: ReactElement,
  renderOptions?: RenderOptions,
) => {
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export const renderHookWithIntl = <P, R>(hook: (props: P) => R) => {
  return renderHook(hook, { wrapper: Wrapper });
};
