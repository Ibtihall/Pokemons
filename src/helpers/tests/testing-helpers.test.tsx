import * as renderHelpers from '@testing-library/react';
import { renderWithIntl, Wrapper } from '../testing-helpers';

import React from 'react';

afterEach(renderHelpers.cleanup);

describe('renderWithIntl', () => {
  it('should call render and return rendered component', () => {
    jest.mock('@testing-library/react', () => () => ({ render: jest.fn() }));

    const { asFragment } = renderWithIntl(<h2>fake-render</h2>);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Wrapper', () => {
  it('should render with defined wrapped elements', () => {
    const { asFragment } = renderHelpers.render(
      //@ts-ignore
      <Wrapper>
        <h1>fake-wrapper</h1>
      </Wrapper>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
