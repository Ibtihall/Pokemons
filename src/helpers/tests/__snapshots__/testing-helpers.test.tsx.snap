// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should render with defined wrapped elements 1`] = `
<DocumentFragment>
  <h1>
    fake-wrapper
  </h1>
</DocumentFragment>
`;

exports[`renderWithIntl should call render and return rendered component 1`] = `
<DocumentFragment>
  <h2>
    fake-render
  </h2>
</DocumentFragment>
`;
